pkg/monitor/monitor_test.go

```diff
@@ -588,6 +589,44 @@ func testDuplicateOfflineWithSameIPAddress(test *testing.T) {
 	require.NoError(test, err, "pod should not get deleted as there is another node with same IP which is online")
 }
 
+func testDirectAccessVolume(t *testing.T) {
+
+	setupWithNewMockDriver(t)
+
+	defer teardownWithNewMockDriver(t)
+
+	pod := newPod("driverPod", []string{driverVolumeName}, nodeForPod)
+	_, err := core.Instance().CreatePod(pod)
+	require.NoError(t, err, "failed to create pod")
+
+	noStoragePod := newPod("noStoragePod", nil, nodeForPod)
+	_, err = core.Instance().CreatePod(noStoragePod)
+	require.NoError(t, err, "failed to create pod")
+
+	driverNodes := getDriverNodes(len(nodes.Items))
+	driverNodes[0].Status = volume.NodeOffline
+	volumeDriver.EXPECT().GetNodes().Return(driverNodes, nil).AnyTimes()
+	volumes := []*volume.Info{
+		{
+			VolumeName:     "volume1",
+			DataNodes:      []string{driverNodes[0].StorageID},
+			DirectAttached: true,
+		},
+	}
+
+	wffcVolumes := make([]*volume.Info, 0)
+
+	volumeDriver.EXPECT().InspectNode("node1").Return(driverNodes[0], nil).AnyTimes()
+	volumeDriver.EXPECT().GetCSIPodPrefix().Return("px-csi-ext-", nil).AnyTimes()
+	volumeDriver.EXPECT().GetPodVolumes(&pod.Spec, pod.Namespace, false).Return(volumes, wffcVolumes, nil).AnyTimes()
+	volumeDriver.EXPECT().GetPodVolumes(&noStoragePod.Spec, noStoragePod.Namespace, false).Return(wffcVolumes, wffcVolumes, nil).AnyTimes()
+
+	testNodeOfflineTimeout = 95 * time.Second
+	time.Sleep(testNodeOfflineTimeout)
+	_, err = core.Instance().GetPodByName(pod.Name, "")
+	require.NoError(t, err, "pod should not get deleted as there is another node with same IP which is online")
```

Comment:
Do we have another node in this test with same IP?

Reply:
I guess we should mention here that pod with volume `volume1` is not deleted since it is directly attached while the pod without any volume gets deleted? Please correct if I am wrong.

Reply:
Fixed.